modules/core/04-channel/keeper/upgrade.go

```diff
@@ -326,3 +325,61 @@ func (k Keeper) constructProposedUpgrade(ctx sdk.Context, portID, channelID stri
 		LatestSequenceSend: nextSequenceSend - 1,
 	}, nil
 }
+
+// abortHandshake will restore the channel state and flush status to their pre-upgrade state so that upgrade is aborted.
+// any unnecessary state is deleted. An error receipt is written, and the OnChanUpgradeRestore callback is called.
+func (k Keeper) abortHandshake(ctx sdk.Context, portID, channelID string, upgradeError *types.UpgradeError) error {
```

Comment:
discussed with @chatton about this API yesterday - I think maybe it needs to accept a built-in `error` instead of `types.UpgradeError` as we can't make any guarantees that applications will always return `types.UpgradeError`

also nit: rename to `AbortUpgrade` rather than handshake as that has spill over with the opening channel handshake naming